modules/redshift-loader/src/main/scala/com/snowplowanalytics/snowplow/loader/redshift/Redshift.scala

```diff
@@ -85,11 +85,12 @@ object Redshift {
             discovery: DataDiscovery,
             eventTableColumns: EventTableColumns,
             i: Unit,
-            disableMigration: List[SchemaCriterion]
+            disableMigration: List[SchemaCriterion],
+            legacyPartitioning: Boolean
           ): LoadStatements = {
             val shreddedStatements = discovery.shreddedTypes
               .filterNot(_.isAtomic)
-              .groupBy(_.getLoadPath)
+              .groupBy(_.getLoadPath(legacyPartitioning))
```

Comment:
I _think_ lines 97-101 are now incorrect for the case when legacy partitioning is enabled and if the schema requires a recovery table.

If legacy partitioning is disabled, then it is OK.  And if schema evolution rules are not broken, then it is OK.

Imagine this batch uses versions `1-0-0` and `1-0-1` of the same schema.  And imagine `1-0-1` is broken so needs a recovery table.

When you call `.map(_.head)` you do not know whether the `head` is `1-0-0` or `1-0-1`.  So when you call `mergeResult = ....` (line 97) you don't know if you get the good schema or the recovery schema.

---

Before I offer a solution... I need to understand the corresponding feature in the transformer, #1315 .  I have not yet read through that PR, so shall do so next.